meta-mentor-staging/classes/sdk_auto_relocate.bbclass

```diff
@@ -17,23 +17,31 @@ SDK_POST_INSTALL_COMMAND_append = 'echo "${env_setup_script%/*}" >"${env_setup_s
 sdkpath_to_bindir = "${@os.path.relpath('${SDKPATHNATIVE}${bindir_nativesdk}', '${SDKPATH}')}"
 
 toolchain_env_script_reloc_fragment () {
-    cat <<END >>"$script"
+    mv "$script" "$script.fragment"
+    cat >"$script" <<END
 if [ -z "\$SDK_RELOCATING" ]; then
     if [ -n "\$BASH_SOURCE" ] || [ -n "\$ZSH_NAME" ]; then
         if [ -n "\$BASH_SOURCE" ]; then
             scriptdir="\$(cd "\$(dirname "\$BASH_SOURCE")" && pwd)"
         elif [ -n "\$ZSH_NAME" ]; then
             scriptdir="\$(cd "\$(dirname "\$0")" && pwd)"
         fi
-        env -i "\$scriptdir/${sdkpath_to_bindir}/sdk-auto-relocate" && SDK_RELOCATING=1 . "\$scriptdir/environment-setup-${REAL_MULTIMACH_TARGET_SYS}"
-        unset scriptdir
+
+        if [ -e "\$scriptdir/${sdkpath_to_bindir}/sdk-auto-relocate" ]; then
+            env -i "\$scriptdir/${sdkpath_to_bindir}/sdk-auto-relocate" && SDK_RELOCATING=1 . "\$scriptdir/environment-setup-${REAL_MULTIMACH_TARGET_SYS}"
+        else
+            echo >&2 "Warning: Unable to find sdk-auto-relocate script"
+        fi
     else
         echo >&2 "Warning: Unable to determine SDK install path from environment setup script location. Please run <installdir>/${sdkpath_to_bindir}/sdk-auto-relocate manually."
+        scriptdir="\$(pwd)"
```

Comment:
I think we might be better off just not doing the sed if we don't know the correct scriptdir, rather than setting it to $PWD and then sed'ing SDKPATH to that. Thoughts?